iml-agent/src/rpm.rs

```diff
@@ -0,0 +1,25 @@
+use regex::Regex;
+
+use std::process::Output;
+
+use crate::{agent_error::ImlAgentError, cmd::cmd_output};
+
+fn parse(output: Output) -> Result<bool, ImlAgentError> {
+    if output.status.success() {
+        Ok(true)
+    } else {
+        let stdout = output.stdout.clone();
+        let re = Regex::new(r"^package .*? is not installed\n$").unwrap();
+        let s = String::from_utf8(stdout)?;
+        if re.is_match(&s) {
+            Ok(false)
+        } else {
+            Err(ImlAgentError::CmdOutputError(output))
+        }
+    }
+}
+
+pub(crate) async fn query(package_name: &str) -> Result<bool, ImlAgentError> {
```

Comment:
Perhaps change the name of this fn so it's return type is more clear:

```suggestion
pub(crate) async fn installed(package_name: &str) -> Result<bool, ImlAgentError> {
```

Reply:
Done